src/openexr.imageio/exroutput.cpp

```diff
@@ -290,6 +290,71 @@ set_exr_threads()
 
 
 
+constexpr float ACES_AP0_chromaticities[8] = {
+    0.7347f,  0.2653f,  // red
+    0.0f,     1.0f,     // green
+    0.0001f,  -0.077f,  // blue
+    0.32168f, 0.33767f  // white
+};
+
+
+
+bool
+is_aces_container_compliant(const OIIO::ImageSpec& spec)
+{
+    // Check channels
+    std::vector<std::string> allowed_sets
+        = { "B,G,R",
+            "A,B,G,R"
+            "B,G,R,left.B,left.G,left.R",
+            "A,B,G,R,left.A,left.B,left.G,left.R" };
+    std::string channels;
+    for (int c = 0; c < spec.nchannels; ++c) {
+        if (c > 0)
+            channels += ",";
+        channels += spec.channelnames[c];
+    }
+    if (std::find(allowed_sets.begin(), allowed_sets.end(), channels)
+        == allowed_sets.end())
+        return false;
+
+    // Check data type
+    if (spec.format != OIIO::TypeDesc::HALF)
+        return false;
+
+    // Check compression
+    std::string compression = spec.get_string_attribute("compression", "zip");
+    if (compression != "none")
+        return false;
+
+    return true;
+}
+
+
+
+bool
+process_aces_container(OIIO::ImageSpec& spec, std::string mode)
+{
+    bool is_compliant = is_aces_container_compliant(spec);
+
+    if (!is_compliant) {
+        std::cerr << "WARNING: Image spec is not ACES Container compliant\n";
```

Comment:
It's usually not desirable for a library to write to stderr (or stdout), since you have no idea what kind of app you're being called from. (For example, for Windows GUI apps, I think stderr goes nowhere.) We prefer to do all error reporting through calls to ImageInput::errorfmt() so that the calling app can retrieve them and respond in a way that's idiomatic for that app.

Unfortunately, we don't have a way currently to report warnings. We'll probably add that in the future, but for now, report the errors (the official way) and just leave a comment specifying where warning should happen, in the future when they are supported.